src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,86 @@
+import * as _ from 'lodash'
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+
+import TreeListItem from './TreeListItem'
+import { UIComponent, childrenExist, createShorthandFactory } from '../../lib'
+import { ShorthandValue, ShorthandRenderFunction } from '../../../types/utils'
+import { Accessibility } from '../../lib/accessibility/types'
+import { defaultBehavior } from '../../lib/accessibility'
+import { commonUIComponentPropTypes, childrenComponentPropTypes } from '../../lib/commonPropTypes'
+import { UIComponentProps, ChildrenComponentProps } from '../../lib/commonPropInterfaces'
+import * as customPropTypes from '../../lib/customPropTypes'
+
+export interface TreeProps extends UIComponentProps<any, any>, ChildrenComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * @default defaultBehavior
+   */
+  accessibility?: Accessibility
+
+  /** Shorthand array of props for Tree. */
+  items: ShorthandValue[]
+
+  /** Whether the tree is a subtree. */
+  nested?: boolean
+
+  /**
+   * A custom render function the title slot.
+   *
+   * @param {React.ReactType} Component - The computed component for this slot.
+   * @param {object} props - The computed props for this slot.
+   * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+   */
+  renderTitle?: ShorthandRenderFunction
```

Comment:
would suggest to rename to `renderItemTitle`, as it will better reflect semantics of this prop